training/03_Infrastructure/02_Advanced/Adding_Users.adoc

```diff
@@ -94,8 +94,100 @@ lab_users:
 [...]
 ----
 
+=== Calling the right task
+We've just added the task and users, but we have not called it yet. In order to make the task to be called, we need to place it in the right place
+
+[source,bash]
+----
+- name: Setup lab-user
+  hosts: all
+  become: true
+  tasks:
+    - name: Setup lab-user
+      include_tasks: ./setup_lab_user.yml
+----
+
 == Setting up ssh keys
 Even if users have been created, no ssh keys have been defined, so we have not clue on how to log into the nodes machines, as we do not have a password.
 
 We want to be able to create ssh keys for our users and we need to be able to let our users to access them, easily (in case someone wants to re use our config).
 
+SSH keys configuration tasks are also listed on the `setup_lab_user.yml` file. Again, as we did with users, please add loops for all of the tasks (final result is shown below):
+
+[source,bash]
+----
+[agilpipp-redhat.com@bastion configs]$ cd a-base-config/
+[agilpipp-redhat.com@bastion a-base-config]$ vi setup_lab_user.yml
+[...]
+- name: create /home/{{ __ssh_dir }}/.ssh
+  file:
+    dest: /home/{{ __ssh_dir }}/.ssh
+    mode: 0700
+    owner: "{{ __ssh_dir }}"
+    group: "users"
+    state: directory
+  loop: "{{ lab_users }}"
+  loop_control:
+    loop_var: __ssh_dir
+
+- name: copy the environment .pem key
+  copy:
+    src: "{{ output_dir }}/{{ env_authorized_key }}"
+    dest: "/home/{{ __ssh_pem }}/.ssh/{{ env_authorized_key }}.pem"
+    owner: "{{ __ssh_pem }}"
+    group: "users"
+    mode: 0400
+  loop: "{{ lab_users }}"
+  loop_control:
+    loop_var: __ssh_pem
+
+- name: copy the environment .pub key
+  copy:
+    src: "{{ output_dir }}/{{ env_authorized_key }}.pub"
+    dest: "/home/{{ __ssh_pub }}/.ssh/{{ env_authorized_key }}.pub"
+    owner: "{{ __ssh_pub }}"
+    group: "users"
+    mode: 0400
+  loop: "{{ lab_users }}"
+  loop_control:
+    loop_var: __ssh_pub
+
+- name: copy .ssh/config template
+  template:
+    src: ./files/ssh_config.j2
+    dest: /home/{{ __ssh_conf }}/.ssh/config
+    owner: "{{ __ssh_conf }}"
+    group: "users"
+    mode: 0400
+  loop: "{{ lab_users }}"
+  loop_control:
+    loop_var: __ssh_conf
```

Comment:
This is looping over `lab_users` but setting a variable `__ssh_conf`? Shouldn't the variable be `__lab_user`?

Reply:
__lab_user being defined instead of item so we can later access it an use it